crates/xtask/src/utils.rs

```diff
@@ -0,0 +1,117 @@
+use ansi_term::Colour::White;
+use anyhow::{anyhow, Context, Result};
+use camino::Utf8PathBuf;
+use cargo_metadata::MetadataCommand;
+use lazy_static::lazy_static;
+
+use std::{
+    collections::HashMap,
+    convert::TryFrom,
+    env,
+    process::{Command, Output, Stdio},
+    str,
+};
+
+const MANIFEST_DIR: &str = env!("CARGO_MANIFEST_DIR");
+
+lazy_static! {
+    pub(crate) static ref PKG_VERSION: String =
+        rover_version().expect("Could not find Rover's version.");
+}
+
+pub(crate) fn info(msg: &str) {
+    let info_prefix = White.bold().paint("info:");
+    eprintln!("{} {}", &info_prefix, msg);
+}
+
+pub(crate) fn rover_version() -> Result<String> {
+    let project_root = project_root()?;
+    let metadata = MetadataCommand::new()
+        .manifest_path(project_root.join("Cargo.toml"))
+        .exec()?;
+
+    Ok(metadata
+        .root_package()
+        .ok_or_else(|| anyhow!("Could not find root package."))?
+        .version
+        .to_string())
+}
+
+pub(crate) fn project_root() -> Result<Utf8PathBuf> {
+    let manifest_dir = Utf8PathBuf::try_from(MANIFEST_DIR)
+        .with_context(|| "Could not find the root directory.")?;
+    let root_dir = manifest_dir
+        .ancestors()
+        .nth(2)
+        .ok_or_else(|| anyhow!("Could not find project root."))?;
+    Ok(root_dir.to_path_buf())
+}
+
+pub(crate) fn exec(
+    command_name: &str,
+    args: &[&str],
+    directory: &Utf8PathBuf,
+    verbose: bool,
+    env: Option<HashMap<String, String>>,
+) -> Result<CommandOutput> {
+    if Command::new(command_name)
+        .arg("--version")
+        .stdout(Stdio::null())
+        .status()
+        .is_err()
+    {
+        return Err(anyhow!(
+            "You must have `{}` installed to run this command.",
+            command_name
+        ));
+    }
```

Comment:
This fails in PowerShell. I am trying to debug atm and am wondering if we are writing to the PATH env var somewhere in `cargo xtask prep` command?

Reply:
hmmm.... no i don't think so. i didn't test this at all on windows. i'm guessing it's because there's no `.exe` suffix. i think if i just add an `if cfg!(windows)` and add a `.exe` to the `command_name` before passing to `Command::new` it should maybe work?